src/ucs/algorithm/crc.c

```diff
@@ -4,39 +4,66 @@
  * See file LICENSE for terms.
  */
 
-#include "crc.h"
+#include <ucs/algorithm/crc.h>
 
 #include <string.h>
 
-
-uint16_t ucs_crc16(const void *buffer, size_t size)
+/* This is CRC-16-CCITT with reversed (0x8408) polynomial representation */
+uint16_t ucs_crc16(uint16_t prev_crc, const void *buffer, size_t size)
 {
+    
+    const uint8_t *end = (const uint8_t*)(buffer + size);
+    uint16_t crc = ~prev_crc;
     const uint8_t *p;
-    uint16_t result;
-    uint8_t data;
-    int bit;
+    uint8_t data, bit;
 
     if (size == 0) {
-        return 0;
+        return crc;
```

Comment:
why do we need/use prev_crc?

Reply:
this was added to align the crc16 with crc32. do you think it is better to avoid this?

Reply:
imo, yes

Reply:
done